File: aspnet-version/backend/PetConnectAlmost/Models/Pet.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PetConnectAlmost.Models
{
    [Table("pets")]
    public class Pet
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Column("user_id")]
        public long UserId { get; set; }

        [Required]
        [Column("pet_name")]
        public string PetName { get; set; }

        [Required]
        [Column("species")]
        public string Species { get; set; }

        [Column("breed")]
        public string? Breed { get; set; }

        [Column("age")]
        public int? Age { get; set; }

        [Column("image_url")]
        public string? ImageUrl { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; }

        // Navigation
        [ForeignKey("UserId")]
        public virtual User User { get; set; }
    }
}

File: springboot-version/backend/src/main/java/com/petconnect/entity/Pet.java
package com.petconnect.entity;

import jakarta.persistence.Entity;

@Entity
@jakarta.persistence.Table(name = "pets")
@lombok.Data
@lombok.NoArgsConstructor
@lombok.AllArgsConstructor
public class Pet {
    @jakarta.persistence.Id
    @jakarta.persistence.GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    private Long id;

    @jakarta.persistence.Column(nullable = false)
    private String petName;

    @jakarta.persistence.Column(nullable = false)
    private String species;

    private String breed;
    private Integer age;
    private String imageUrl;

    @jakarta.persistence.Column(name = "created_at", updatable = false)
    private java.time.LocalDateTime createdAt;

    @jakarta.persistence.Column(name = "updated_at")
    private java.time.LocalDateTime updatedAt;

    @jakarta.persistence.ManyToOne
    @jakarta.persistence.JoinColumn(name = "user_id", nullable = false)
    @org.hibernate.annotations.OnDelete(action = org.hibernate.annotations.OnDeleteAction.CASCADE)
    private User user;

    @jakarta.persistence.PrePersist
    protected void onCreate() {
        createdAt = java.time.LocalDateTime.now();
        updatedAt = java.time.LocalDateTime.now();
    }

    @jakarta.persistence.PreUpdate
    protected void onUpdate() {
        updatedAt = java.time.LocalDateTime.now();
    }
}
